cli/lsp/diagnostics.rs

```diff
@@ -358,14 +378,15 @@ pub struct DiagnosticsServer {
   performance: Arc<Performance>,
   ts_server: Arc<TsServer>,
   batch_counter: DiagnosticBatchCounter,
-  state: Arc<RwLock<DiagnosticsState>>,
+  state: DiagnosticsState,
 }
 
 impl DiagnosticsServer {
   pub fn new(
     client: Client,
     performance: Arc<Performance>,
     ts_server: Arc<TsServer>,
+    state: DiagnosticsState,
```

Comment:
Nitpick and something for debate: I've been slowly changing things to move the `Arc` to a higher level (so, `Arc<DiagnosticsState>` similar to `Arc<Performance>` and `Arc<TsServer>`) because then it's very clear that cloning will clone a reference to the struct and it's clear that it's shared amongst other code (I think it being in an `Arc` is a higher level concern).